modules/afprog/afprog.c

```diff
@@ -358,18 +430,55 @@ afprogram_dd_init(LogPipe *s)
     }
   log_pipe_append(&self->super.super.super, (LogPipe *) self->writer);
 
-  return afprogram_dd_reopen(self);
+  return restore_successful ? TRUE : afprogram_dd_reopen(self);
+}
+
+static inline void
+afprogram_dd_store_reload_info(AFProgramDestDriver * const self, GlobalConfig * const cfg)
+{
```

Comment:
I don't really see the added value in specifying that self and cfg as a
pointer is const. It is an implementation detail of the function and
doesn't need visibility outside.

We are not using this convention anywhere and I wouldn't start it now. I
can be convinced of course (as always), but there should be a tangible
benefit.


-- 
Bazsi

On Mon, Oct 5, 2015 at 5:58 PM, MÓZES Ádám István <notifications@github.com>
wrote:

> In modules/afprog/afprog.c
> <https://github.com/balabit/syslog-ng/pull/722#discussion_r41161499>:
>
> > @@ -358,18 +430,55 @@ afprogram_dd_init(LogPipe *s)
> >      }
> >    log_pipe_append(&self->super.super.super, (LogPipe *) self->writer);
> >
> > -  return afprogram_dd_reopen(self);
> > +  return restore_successful ? TRUE : afprogram_dd_reopen(self);
> > +}
> > +
> > +static inline void
> > +afprogram_dd_store_reload_info(AFProgramDestDriver * const self, GlobalConfig * const cfg)
> > +{
>
> The C++ const convention
> <https://en.wikipedia.org/wiki/Const_%28computer_programming%29#C.2B.2B_convention>
> is used here, so in this case the pointer is qualified as a const, not to
> the referenced type.
> On the other hand, I would attach the asterisk to the type, but this is
> not common in the current codebase.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/722/files#r41161499>.
>